### PR TITLE
Backend of deployable chunk 1 of owner invitations

### PR DESCRIPTION
**Pull request by carols10cents**
This adds a table for storing pending crate owner invitations and a backend route for listing the current user's invitations. 

I'll probably merge this later today if no one has time to review so that @natboehm can build off of this tomorrow :)

### CODE REVIEW

**Comment by vignesh-sankaran** [comment]
I'm not going to have the time to review this, I do like how you've split out the tests into `owners.rs`, keeps the file sizes smaller :)

**Comment by carols10cents** [comment]
bors: r+

File: src/crate_owner_invitation.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+/// Handles the `GET /me/crate_owner_invitations` route.
+pub fn list(req: &mut Request) -> CargoResult<Response> {
+    let conn = &*req.db_conn()?;
+    let user_id = req.user()?.id;
+
+    let invitations = crate_owner_invitations::table
+        .filter(crate_owner_invitations::invited_user_id.eq(user_id))
+        .load::<CrateOwnerInvitation>(&*conn)?
+        .into_iter()
+        .map(|i| i.encodable(conn))
```
Comment (sgrif) [inline src/crate_owner_invitation.rs]: This has an N+1 queries bug, doesn't it? Should we be selecting the crate and owner name along with the invitation?

Comment (carols10cents) [reply src/crate_owner_invitation.rs]: Oops. I'll fix it before we start linking to this page.

**Pull request merged**